sdk/identity/azure-identity/azure/identity/_exceptions.py

```diff
@@ -0,0 +1,9 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+from azure.core.exceptions import AzureError
+
+
+class CredentialUnavailableError(AzureError):
```

Comment:
CredentialUnavailableError needs to be a sub-class of ClientAuthenticationError. Or it will be a breaking change.

Reply:
ClientAuthenticationError inherits HttpResponseError, which is misleading as a base here because CredentialUnavailableError is raised when no request was sent.

I agree changing the hierarchy is a larger issue. For this PR I'll have CredentialUnavailableError inherit ClientAuthenticationError, and we can revisit the hierarchy later.